scripts/scil_fibertube_compute_density.py

```diff
@@ -0,0 +1,161 @@
+                        'as data_per_streamline. \n'
+                        'For both of these requirements, see \n'
+                        'scil_tractogram_filter_collisions.py.')
+
+    p.add_argument('--out_density_map', default=None, type=str,
+                   help='Path of the output volumetric density image file')
+
+    p.add_argument('--out_density_measures', default=None, type=str,
+                   help='Path of the output file containing central \n'
+                        'tendency measures about volumetric density. \n'
+                        '(Must be .json)')
+
+    p.add_argument('--out_collision_map', default=None, type=str,
+                   help='Path of the output collision density image file')
+
+    p.add_argument('--out_collision_measures', default=None, type=str,
+                   help='Path of the output file containing central \n'
+                        'tendency measures about collision density. \n'
+                        '(Must be .json)')
+
+    add_overwrite_arg(p)
+    add_verbose_arg(p)
+    add_json_args(p)
+
+    return p
+
+
+def main():
+    parser = _build_arg_parser()
+    args = parser.parse_args()
+
+    logging.getLogger().setLevel(logging.getLevelName(args.verbose))
+    logging.getLogger('numba').setLevel(logging.WARNING)
+
+    if not (args.out_density_map or args.out_density_measures or
+            args.out_collision_map or args.out_collision_measures):
+        parser.error('No output argument given')
+
+    if os.path.splitext(args.in_fibertubes)[1] != '.trk':
+        parser.error('Invalid input streamline file format (must be trk):' +
+                     '{0}'.format(args.in_fibertubes))
+
+    if args.out_density_measures:
+        if os.path.splitext(args.out_density_measures)[1] != '.json':
+            parser.error('Invalid output file format (must be json): {0}'
+                         .format(args.out_density_measures))
+
+    if args.out_collision_measures:
+        if os.path.splitext(args.out_collision_measures)[1] != '.json':
+            parser.error('Invalid output file format (must be json): {0}'
+                         .format(args.out_collision_measures))
+
+    assert_inputs_exist(parser, args.in_fibertubes)
+    assert_outputs_exist(parser, args, [],
+                         [args.out_density_map, args.out_density_measures,
+                          args.out_collision_map, args.out_collision_measures])
+
+    logging.debug('Loading tractogram & diameters')
+    sft = load_tractogram(args.in_fibertubes, 'same')
+    sft.to_voxmm()
+    sft.to_center()
+
+    if "diameters" not in sft.data_per_streamline:
+        parser.error('No diameters found as data per streamline on ' +
+                     args.in_fibertubes)
+
+    logging.debug('Computing fibertube density')
+    (density_grid,
+     density_flat,
+     collision_grid,
+     collision_flat) = fibertube_density(sft, 10,
+                                         args.verbose == 'WARNING')
```

Comment:
This formatting is a bit ugly no? @arnaudbore

Reply:
I agree

Reply:
The part after = fits on a single line.